### PR TITLE
Add JointSet::get_mut and get_unknown_gen_mut

### PR DESCRIPTION
**Pull request by emilk**
Fix https://github.com/dimforge/rapier/issues/121

### CODE REVIEW

File: src/dynamics/joint/joint_set.rs [hunk 1]
```diff
@@ -76,6 +76,12 @@ impl JointSet {
         self.joint_graph.graph.edge_weight(*id)
     }
 
+    /// Gets a mutable reference to the joint with the given handle.
+    pub fn get_mut(&mut self, handle: JointHandle) -> Option<&mut Joint> {
+        let id = self.joint_ids.get_mut(handle.0)?;
```
Comment (sebcrozet) [inline src/dynamics/joint/joint_set.rs:81]: This does not have to be mutable:

```suggestion
        let id = self.joint_ids.get(handle.0)?;
```

File: src/dynamics/joint/joint_set.rs [hunk 2]
```diff
@@ -93,6 +99,23 @@ impl JointSet {
...
+    ///
+    /// This is useful when you know you want the joint at position `i` but
+    /// don't know what is its current generation number. Generation numbers are
+    /// used to protect from the ABA problem because the joint position `i`
+    /// are recycled between two insertion and a removal.
+    ///
+    /// Using this is discouraged in favor of `self.get_mut(handle)` which does not
+    /// suffer form the ABA problem.
+    pub fn get_unknown_gen_mut(&mut self, i: usize) -> Option<(&mut Joint, JointHandle)> {
+        let (id, handle) = self.joint_ids.get_unknown_gen_mut(i)?;
```
Comment (sebcrozet) [inline src/dynamics/joint/joint_set.rs:112]: This does not have to be mutable:

```suggestion
        let (id, handle) = self.joint_ids.get_unknown_gen(i)?;
```

**Review by sebcrozet — CHANGES_REQUESTED** [review]
Thank you for this PR!

File: src/dynamics/joint/joint_set.rs (hunk 1 above)
Comment (emilk) [reply src/dynamics/joint/joint_set.rs:81]: ah no, of course

**Review by sebcrozet — APPROVED** [review]

**Comment by sebcrozet** [comment]
Thanks!

**Pull request merged**